Casbin/Model/Assertion.cs

```diff
@@ -237,5 +237,55 @@ int PolicyComparison(IPolicyValues p1, IPolicyValues p2)
             _policy.Sort(PolicyComparison);
             return true;
         }
+
+        private bool TryGetSubjectHierarchyDomainIndex(out int index)
+        {
+            if (Tokens is null)
+            {
+                index = -1;
+                return false;
+            }
+            return Tokens.TryGetValue("dom", out index);
+        }
+
+        private bool TryGetSubjectHierarchySubjectIndex(out int index)
+        {
+            if (Tokens is null)
+            {
+                index = -1;
+                return false;
+            }
+            return Tokens.TryGetValue("sub", out index);
+        }
+
+
+        internal bool TrySortPoliciesBySubjectHierarchy(Dictionary<string, int> subjectHierarchyMap, Func<string, string, string> nameFormatter)
+        {
+            if(TryGetSubjectHierarchyDomainIndex(out int domainIndex) is false)
+            {
+                domainIndex = -1;
+            }
+            if (TryGetSubjectHierarchySubjectIndex(out int subjectIndex) is false)
+            {
+                subjectIndex = -1;
```

Comment:
An exception may be thrown when sorting the policy if the index is -1, we can directly return false here.